lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -1756,19 +1756,32 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
                 rewriter.create<Torch::AtenEqIntOp>(binder.getLoc(), dim, zero);
             isZero =
                 rewriter.create<Torch::AtenIntBoolOp>(binder.getLoc(), isZero);
-            Value adjustment = zero;
-            int64_t inputDimsSize = dataSizes.size();
-            if (i < inputDimsSize) {
-              adjustment = rewriter.create<Torch::ConstantIntOp>(
+
+            int64_t dataRank = dataSizes.size();
+            if (i < dataRank) {
+              auto torchIntTy = rewriter.getType<Torch::IntType>();
+              auto int64Ty = rewriter.getIntegerType(64, true);
+              auto dimTy = rewriter.getType<Torch::ValueTensorType>(
+                  ArrayRef<int64_t>(), int64Ty);
+              auto boolTy = rewriter.getType<Torch::ValueTensorType>(
+                  ArrayRef<int64_t>(), rewriter.getI1Type());
+              Value iv = rewriter.create<Torch::ConstantIntOp>(
+                  binder.getLoc(), rewriter.getI64IntegerAttr(i));
+              Value inDim = rewriter.create<Torch::AtenSizeIntOp>(
+                  binder.getLoc(), torchIntTy, data, iv);
+              isZero = rewriter.create<Torch::PrimNumToTensorScalarOp>(
+                  binder.getLoc(), boolTy, isZero);
+              inDim = rewriter.create<Torch::PrimNumToTensorScalarOp>(
+                  binder.getLoc(), dimTy, inDim);
+              dim = rewriter.create<Torch::PrimNumToTensorScalarOp>(
+                  binder.getLoc(), dimTy, dim);
+              Value finalDim = rewriter.create<Torch::AtenWhereSelfOp>(
```

Comment:
The codebase generally uses atenIntBoolOp to convert the bool into an int 

and then use an AtenMulOp to add it.

This is more efficient than that right?

Reply:
Should someone go through the other ones and try to fix them to use AtenWhereSelfOp too?

Reply:
This is less about efficiency and more about folding. I really want an atenSelectScalar op that takes all scalars but sadly it doesn't exist so we have to proxy with the scalar-tensor version. Multiplications with bools and additions is a way to simulate a select operation but many cpus have instructions to make ternary selection easier.